apollo-router/src/router.rs

```diff
@@ -221,6 +221,8 @@ impl SchemaSource {
                 urls,
                 poll_interval,
             } => {
+                // With regards to ELv2 licensing, the code inside this block
+                // is license key functionality
```

Comment:
It seems unusual to apply a license to part of a file. Is that a valid thing to do?

Reply:
As far as I understand the entire file (like the rest of the project) is under the same overall license, but the special "license key-related" provisions of the license apply only to this part.

Reply:
@SimonSapin's interpretation is correct.